contexts/WeatherUnit.js

```diff
@@ -1,14 +1,64 @@
-import { createContext, useContext } from "react";
+import { createContext, useContext, useReducer } from "react";
+import PropTypes from "prop-types";
+import { useTranslation } from "react-i18next";
+
+export const defaultTempUnit = "celsius";
+export const defaultWindspeedUnit = "m";
 
 export const defaultUnits = {
-  tempUnit: "C",
-  windspeedUnit: "m",
+  tempUnit: defaultTempUnit,
+  windspeedUnit: defaultWindspeedUnit,
+};
+
+const WeatherUnitContext = createContext();
+
+const unitReducer = (state, action) => {
+  const { type } = action;
+
+  switch (type) {
+    case "fahrenheit":
+      return { ...state, tempUnit: "fahrenheit" };
+    case "celsius":
+      return { ...state, tempUnit: "celsius" };
+    case "NM":
+      return { ...state, windspeedUnit: "NM" };
+    case "mi":
+      return { ...state, windspeedUnit: "mi" };
+    case "m":
+      return { ...state, windspeedUnit: "m" };
+    default:
+      throw new Error(`Unhandled action type: ${type}`);
+  }
 };
 
-const WeatherUnitContext = createContext(defaultUnits);
+export const WeatherUnitProvider = ({ children }) => {
```

Comment:
did a lot of work on the weather unit provider. Originally the provider was storing values set by the state in the summit status block, but with all of the modals having a unit localization control that means a lot of passing callbacks and props around. 

Instead I opted to move the state management to the provider itself and use a reducer to track the current units. Any instance of the unit localization component will receive the state and a dispatch function that can be used to set the unit and keep callbacks + re-renders to a minimum